src/dicomslide/matrix.py

```diff
@@ -253,7 +253,8 @@ def __init__(
         if self._metadata.SamplesPerPixel == 3 and correct_color:
             if (
                 hasattr(self._metadata, 'OpticalPathSequence') and
-                hasattr(self._metadata.OpticalPathSequence[0], 'ICCProfile')
```

Comment:
If the value of the Samples Per Pixel attribute is `3`, then there must be an ICC Profile attribute. If that's not the case, then I suggest raising an error.